.github/workflows/deploy-js-client.yml

```
@@ -31,37 +36,74 @@ env:
+          else
+            echo type="ref" >> $GITHUB_OUTPUT
+          fi
+
   build_programs:
     name: Programs
     uses: ./.github/workflows/build-programs.yml
     secrets: inherit
+    needs: check_tag
+    if: needs.check_tag.outputs.type == 'ref'
+    with:
+      git_ref: ${{ inputs.git_ref }}
```

actionlint (1.7.11)

[error] input "git_ref" is not defined in "./.github/workflows/build-programs.yml" reusable workflow. defined inputs are "rust", "solana"

(workflow-call)